### PR TITLE
Stop exporting `__data_end` by default

### PR DESCRIPTION
**Pull request by sbc100**
Since https://github.com/WebAssembly/binaryen/pull/3300
wasm-emscripten-finalize no longer needs this.

### CODE REVIEW

**Review by kripken — APPROVED** [review]

File: tests/code_size/hello_webgl2_wasm2js.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
   "a.html.gz": 386,
-  "a.js": 21668,
-  "a.js.gz": 8361,
+  "a.js": 21688,
+  "a.js.gz": 8372,
   "a.mem": 3171,
   "a.mem.gz": 2715,
-  "total": 25435,
-  "total_gz": 11462
+  "total": 25427,
```
Comment (tlively) [inline tests/code_size/hello_webgl2_wasm2js.json:8]: How could the size of "a.js" have gone up but the "total" size gone down? Why is "total" undercounting the sizes of "a.html", "a.js", and "a.mem", which add up to 25447? Why did any sizes go up at all? Is that from unrelated changes that were hidden in slack?

**Review by tlively — APPROVED** [review]
LGTM with a satisfying answer to the question of the strange sizes.

File: tests/code_size/hello_webgl2_wasm2js.json (hunk 1 above)
Comment (sbc100) [reply tests/code_size/hello_webgl2_wasm2js.json]: Good question.   Actually its looks like these codesize tests were not actually affected by this change.

I had run `test_minimal_runtime_code_size` with `REBASELINE_TESTS=1` on the assumption that they might change.   Because `test_minimal_runtime_code_size` as a slop factor the results tend to drift by a few bytes over time.. which means that if you run `REBASELINE_TESTS=1` at any given time something like this change will be created.

Its kind of sad because it makes it hard to know precisely what change is responsible for each shift, but its also makes this test a little less fragile.

I think this slop factor might be the explaination for the section question too.  I think pehaps there is bug where REBASELINE_TESTS should ignore slop but is not resulting in slightly strange results here... I'm investigating now. 

Anyway I reverted this part of the change so this is good to land now.

**Comment by DarthGandalf** [comment]
Hi, latest release of binaryen (98) doesn't have that change yet, but emscripten was released with this change already (2.0.9), so it doesn't work unless using latest version of binaryen from git.

(I'm trying to package emscripten for my distro)

**Comment by sbc100** [comment]
Sadly the only supported way to package emscripten today is to build with precise tip-of-tree versions llvm and binaryen.   **Emscripten is not tests against and will likely not work against the stable releases of either llvm or binaryen**.   If you have had success in the past using stable releases then you have been lucky, but its a risky thing to try to do.

For more information see https://github.com/emscripten-core/emscripten/blob/master/docs/packaging.md where we give details about exactly how to find the correct revisions to package.

My recommendation is to include the entire binaryen and llvm sources as part of emscripten source package rather than try to use package dependencies to pull them in.

There are plans to start to supporting stable versions but they are still just plans: #11362 

Also, if you packaging emscripten please consider adding yourself to the list at the bottom of https://github.com/emscripten-core/emscripten/blob/master/docs/packaging.md.

**Comment by DarthGandalf** [comment]
I see, thanks.
